pkg/runtime/runtime.go

```diff
@@ -89,6 +98,16 @@ func (r *Runtime) Start() error {
 	return nil
 }
 
+func (r *Runtime) isReady() bool {
+	// if at least a provider can watch for flags changes, we are ready.
```

Comment:
In my view, we should check for all sync providers 🤔  

The reason is that, as per current ISync implementations, if the sync provider fails at the startup (first fetch), then this results in a flagd startup failure. So if we only check for one sync provider, it might mislead the readiness (this has the potential to impact OFO if we plan to utilize probes there).

Reply:
I had the assumption that, if I configure multiple providers, then they all return the same set of flags. Hence my implementation. Could it be the case that multiple providers see only a partition of the whole FF set?

Reply:
> Could it be the case that multiple providers see only a partition of the whole FF set?

Yep! For instance, a different provider is set on different mounted-files/CRs. This is a common use case, since frequently there's a set of common flags shared by multiple services, but also flags unique to a single service. These may be respresented by different files/CRs, and also even come from different source types.

Reply:
Thanks for the explanation! I will adapt the code accordingly :)